src/mtype.c

```diff
@@ -3980,6 +3980,8 @@ void TypeSArray::resolve(Loc loc, Scope *sc, Expression **pe, Type **pt, Dsymbol
     }
     else
     {
+        if ((*pt)->ty != Terror)
+            next = *pt;     // prevent re-running semantic() on 'next'
```

Comment:
Perhaps you're worried the case that the `resolve()` call happens with error gagging?
That's not a problem. As you see, if `*pt` is Terror, it won't be stored back to `next`, then the error object won't leak out of the gagged scope.